### PR TITLE
Update to Spring 1.5.3.RELEASE and corresponding change to Log4j2

### PR DESCRIPTION
**Pull request by ibaldin**
This is an upgrade to new version of spring. Tested to work with null plugin. Also removed auto-generated files (from jaxb) for configuration parser. 

This also fixes the problem with Java8 where schedule no longer could be retrieved via REST endpoint (fixed by upgrading Spring).

### CODE REVIEW

File: agent/server/na-test-config.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	<tick length="1" unit="minute"/>
 	<plugins>
 		<plugin name="null-plugin">
-			<jar>/Users/ibaldin/workspace-nodeagent2/nodeagent2-git/null-agent/target/null-agent-0.1-SNAPSHOT-jar-with-dependencies.jar</jar>
+			<jar>/Users/ibaldin/workspace-nodeagent2/nodeagent2-git/null-agent/target/null-agent-0.2.1-SNAPSHOT-jar-with-dependencies.jar</jar>
```
Comment (hinchliff) [inline agent/server/na-test-config.xml]: can these be changed to use relative paths?

File: agent/server/na2-spring.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ na2:
 
 # point to log4j configuration
 logging:
-  config: /Users/ibaldin/workspace-nodeagent2/nodeagent2-git/agent/server/log4j-na2.properties
```
Comment (hinchliff) [inline agent/server/na2-spring.yml]: can these be changed to use relative paths?

File: agent/server/log4j2-na2.properties [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+monitoringInterval = 30
+
+appender.console.type = Console
+appender.console.name = STDOUT
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %m%n
+ 
+appender.rolling.type = RollingFile
+appender.rolling.name = RollingFile
+appender.rolling.fileName = /Users/ibaldin/Desktop/na2d.log
```
Comment (hinchliff) [inline agent/server/log4j2-na2.properties]: can these be changed to use relative paths?

**Comment by ibaldin** [comment]
Ready for review again

**Pull request merged**